modules/hikari/src/main/scala/doobie/hikari/Config.scala

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2013-2020 Rob Norris and Contributors
+// This software is licensed under the MIT License (MIT).
+// For more information see LICENSE or https://opensource.org/licenses/MIT
+
+package doobie.hikari
+
+import java.util.Properties
+import java.util.concurrent.{ScheduledExecutorService, ThreadFactory, TimeUnit}
+
+import cats.effect.Sync
+import cats.syntax.show._
+import com.zaxxer.hikari.HikariConfig
+import com.zaxxer.hikari.metrics.MetricsTrackerFactory
+import doobie.enumerated.TransactionIsolation
+import javax.sql.DataSource
+
+import scala.concurrent.duration.Duration
+
+/** Configuration case class, susceptible to PureConfig.
+  * Helps with creating `com.zaxxer.hikari.HikariConfig`,
+  * which in turn is used to create `doobie.hikari.HikariTransactor`.
+  * See the method `HikariTransactor.fromConfig` */
+final case class Config(
+  catalog: Option[String] = None,
+  connectionTimeout: Duration = Duration(30, TimeUnit.SECONDS),
+  idleTimeout: Duration = Duration(10, TimeUnit.MINUTES),
+  leakDetectionThreshold: Duration = Duration.Zero,
+  maximumPoolSize: Option[Int] = None,
+  maxLifetime: Duration = Duration(30, TimeUnit.MINUTES),
+  minimumIdle: Option[Int] = None,
+  password: Option[String] = None,
+  poolName: Option[String] = None,
+  username: Option[String] = None,
+  validationTimeout: Duration = Duration(5, TimeUnit.SECONDS),
+  allowPoolSuspension: Boolean = false,
+  autoCommit: Boolean = true,
+  connectionInitSql: Option[String] = None,
+  connectionTestQuery: Option[String] = None,
+  dataSourceClassName: Option[String] = None,
+  dataSourceJNDI: Option[String] = None,
+  driverClassName: Option[String] = None,
+  initializationFailTimeout: Duration = Duration(1, TimeUnit.MILLISECONDS),
+  isolateInternalQueries: Boolean = false,
+  jdbcUrl: Option[String] = None,
+  readOnly: Boolean = false,
+  registerMbeans: Boolean = false,
+  schema: Option[String] = None,
+  transactionIsolation: Option[TransactionIsolation] = None,
+)
```

Comment:
May I suggest that instead of default parameters we have a `val DEFAULT = Config(..)` in the companion object?

In the tests you can also construct a `Config` from `HikariConfig` (a bit like a round-trip test) which means we won't miss any new fields we add to `Config`.

Reply:
Default parameters are intentionally there. It works well with PureConfig -- which is the whole point of this config case class.
Otherwise I would go with the `val DEFAULT = Config(..)` myself from the start :smile_cat:

Reply:
You'll feel really dirty when doing that once you get used to PureConfig :joy_cat: